packages/taco/test/conditions/compound-condition.test.ts

```diff
@@ -216,4 +216,36 @@ describe('validation', () => {
     });
     expect(condition.value.conditionType).toEqual(CompoundConditionType);
   });
+
+  it('rejects invalid operator', () => {
+    const badObj = {
+      operator: 'invalid-operator',
+      operands: [testContractConditionObj, testTimeConditionObj],
+    };
+    expect(() => new CompoundCondition(badObj)).toThrow();
+  });
+
+  it('rejects empty operands', () => {
+    const badObj = {
+      operator: 'and',
+      operands: [],
+    };
+    expect(() => new CompoundCondition(badObj)).toThrow();
+  });
+
+  it('rejects non-array operands for "and" and "or" operators', () => {
+    const badObj = {
+      operator: 'and',
+      operands: testContractConditionObj,
+    };
+    expect(() => new CompoundCondition(badObj)).toThrow();
+  });
+
+  it('rejects array operands for "not" operator', () => {
```

Comment:
To clarify, the following still holds, correct?
- `not`, `and`, and `or` all accept array operands, just that `not` requires the array to be size 1, and `and`/`or` expect array size to be > 1.
- They all should fail if "operands" is not an array.

Reply:
I've added some tests to cover these cases

Reply:
👍